sdk/tables/Azure.Data.Tables/src/TableConnectionString.cs

```diff
@@ -266,6 +269,44 @@ static Uri CreateUri(string endpoint, string sasToken)
             return false;
         }
 
+        internal static Uri GetSecondaryUriFromPrimary(Uri primaryUri, string accountName = null)
+        {
+            var secondaryUriBuilder = new UriBuilder(primaryUri);
+
+            if (!string.IsNullOrEmpty(accountName))
+            {
+                // We've been provided the accountName, so just insert the '-secondary' suffix after it
+                var indexOfAccountName = secondaryUriBuilder.Host.IndexOf(accountName, StringComparison.OrdinalIgnoreCase);
+                secondaryUriBuilder.Host = secondaryUriBuilder.Host.Insert(indexOfAccountName + accountName.Length, TableConstants.ConnectionStrings.SecondaryLocationAccountSuffix);
+                return secondaryUriBuilder.Uri;
+            }
+
+            var indexOfDot = secondaryUriBuilder.Host.IndexOf('.');
+            if (indexOfDot >= 0 && Uri.CheckHostName(primaryUri.Host) == UriHostNameType.Dns)
+            {
+                // This is a dns name such as contoso.core.windows.net
+                // Insert the '-secondary' suffix after the first part of the host name
+                secondaryUriBuilder.Host = secondaryUriBuilder.Host.Insert(indexOfDot, TableConstants.ConnectionStrings.SecondaryLocationAccountSuffix);
+            }
+            else if (primaryUri.IsLoopback)
```

Comment:
Is it useful to have a secondary uri for the loopback?

Reply:
I assume it matters to Azureite - see https://github.com/Azure/azure-sdk-for-net/issues/17215